src/lib/util/octets.ts

```diff
@@ -0,0 +1,123 @@
+import { Field, checkBitLength } from '../provable/field.js';
+import { UInt8 } from '../provable/int.js';
+import { Provable } from '../provable/provable.js';
+import { Field3, ForeignField } from '../provable/gadgets/foreign-field.js';
+import { assert } from '../provable/gadgets/common.js';
+import { l } from '../provable/gadgets/range-check.js';
+
+export { Octets };
+
+const Octets = {
+  /**
+   * Returns an array of {@link UInt8} elements representing this field element
+   * as little endian ordered bytes.
+   *
+   * If the optional `bytelength` argument is used, it proves that the field
+   * element fits in `bytelength` bytes. The length has to be between 0 and 32,
+   * and the method throws if it isn't.
+   *
+   * **Warning**: The cost of this operation in a zk proof depends on the
+   * `bytelength` you specify, which by default is 32 bytes. Prefer to pass a
+   * smaller `bytelength` if possible.
+   *
+   * @param input - the field element to convert to bytes.
+   * @param bytelength - the number of bytes to fit the element. If the element
+   *                     does not fit in `length` bits, the functions throws an
+   *                     error.
+   *
+   * @return An array of {@link UInt8} element representing this {@link Field} in
+   *         little endian encoding.
+   */
+  fromField(input: Field, bytelength: number = 32): UInt8[] {
+    checkBitLength('Field.toBytes()', bytelength, 32 * 8);
+    if (input.isConstant()) {
+      if (input.toBigInt() >= 1n << (BigInt(bytelength) * 8n)) {
+        throw Error(`toOctets(): ${input} does not fit in ${bytelength} bytes`);
+      }
+      let x = input.toBigInt();
+      return Array.from({ length: bytelength }, (_, k) => new UInt8((x >> BigInt(8 * k)) & 0xffn));
+    }
+    let bytes = Provable.witness(Provable.Array(UInt8, bytelength), () => {
+      let x = input.toBigInt();
+      return Array.from({ length: bytelength }, (_, k) => new UInt8((x >> BigInt(8 * k)) & 0xffn));
+    });
+    let field = bytes
+      .reverse()
+      .map((x) => x.value)
+      .reduce((acc, byte) => acc.mul(256).add(byte));
```

Comment:
the default version of this method with 32 bytes is unsound! since there are several representations of a given field element with 256 bits

Reply:
compare Field to/from bits, where we only allow 254 bits or less

Reply:
You're right, I was using this function only for smaller inputs (limb-length fields actually), but did not add any disclaimer about the soundness for large values when the outcomes are not rangechecked anywhere else. Would you suggest to fix that by adding those checks, limiting it to values which are smaller than the field length, or just remove the `Octets` utility entirely?